controller/utils.go

```diff
@@ -166,81 +178,65 @@ func rollback(
 			}
 
 			waitTimeout := willTimeoutAt.Sub(time.Now())
-			waitCreatingPods(logger, clientset, waitTimeout, configYAML.Name,
-				newlyCreatedPods, nil, mr)
+			waitCreatingAndDeleteOldPods(logger, clientset, redisClient, waitTimeout, configYAML,
+				newlyCreatedPods, nil, nil, nil, mr)
 		}
 	}
 
 	return nil
 }
 
-func createPodsAsTheyAreDeleted(
+func waitTerminatingPods(
 	l logrus.FieldLogger,
-	roomManager models.RoomManager,
-	mr *models.MixedMetricsReporter,
 	clientset kubernetes.Interface,
-	db pginterfaces.DB,
-	redisClient redisinterfaces.RedisClient,
 	timeout time.Duration,
-	configYAML *models.ConfigYAML,
+	namespace string,
 	deletedPods []v1.Pod,
-	scheduler *models.Scheduler,
 	operationManager *models.OperationManager,
-) (createdPods []v1.Pod, timedout, wasCanceled bool) {
+	mr *models.MixedMetricsReporter,
+) (timedout, wasCanceled bool) {
 	logger := l.WithFields(logrus.Fields{
 		"operation": "controller.waitTerminatingPods",
-		"scheduler": configYAML.Name,
+		"scheduler": namespace,
 	})
 
-	createdPods = []v1.Pod{}
-	logger.Debugf("pods to terminate: %#v", names(deletedPods))
+	logger.Debugf("waiting for pods to terminate: %#v", names(deletedPods))
 
 	timeoutTimer := time.NewTimer(timeout)
 	defer timeoutTimer.Stop()
 	ticker := time.NewTicker(1 * time.Second)
 	defer ticker.Stop()
 
-	i := 0
 	for {
 		exit := true
 		select {
 		case <-ticker.C:
-			if operationManager.WasCanceled() {
+			if operationManager != nil && operationManager.WasCanceled() {
```

Comment:
in which scenario operationManager would be nil?

Reply:
operationManger is nil when rolling back (rollback can't be canceled)